### PR TITLE
`Development`: Add database tests run configurations and docs

### PR DESCRIPTION
**Pull request by jakubriegel**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer **locally and on docs page**.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
Despite database server tests for MySQL and Postgres being important part of our CI, Artemis documentation did not included any information on how to start them. IntelliJ run configurations for those tests was also not included  in the repository. This confused some students.

### Description
This PR adds: 
- a documentation on how to start database server tests for MySQL and Postgres
   - docs can be found under _Contributor Guide/Setup Guide/Running database tests locally_ 
- IntelliJ run configurations for starting MySQL and Postgres tests

### Steps for Testing
- read the docs and verify if they are clear (you can download docs as HTML from the artefact produced by the latest _Documentation/docs_ GitHub Action run)
- checkout the branch and run database tests from IntelliJ

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: docs/dev/setup/local-database-tests.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Running database tests locally
+---------------------------------
+
+Prerequisites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Database tests are standard server integration tests executed against real database (MySQL or Postgres) started as Docker containers. Gradle test task will automatically pull necessary images and start Docker containers.
+
+Before running those tests make sure that the following steps were executed:
```
Comment (JanaNF) [inline docs/dev/setup/local-database-tests.rst:11]: Missing "," after "those tests".
Please consider not using passive voice and therefore rephrasing the "Before running..." sentence.

File: docs/dev/setup/local-database-tests.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+1. Install `Docker Desktop <https://docs.docker.com/desktop/#docker-for-mac>`__ or `Docker Engine and Docker CLI <https://docs.docker.com/engine/install/>`__.
+2. Enable `remote access for Docker daemon <https://docs.docker.com/config/daemon/remote-access/>`__.
+
+Executing tests from Intellij
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Database tests can be executed by running of the run configurations provided in the repository:
```
Comment (JanaNF) [inline docs/dev/setup/local-database-tests.rst:19]: Please consider not using passive voice and therefore rephrasing the "Database tests..." sentence.
In the same sentence "of" is not needed. You can remove it.

File: docs/dev/setup/local-database-tests.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+* For MySQL tests: `Artemis Server Tests MySQL`.
+* For Postgres tests: `Artemis Server Tests Postgres`.
+
+Or by creating a custom configuration. Screenshot below presents an example of such run configuration:
```
Comment (JanaNF) [inline docs/dev/setup/local-database-tests.rst:24]: I would change "Or ..." --> to  "Or alternatively, ..."
Add an article to "Screenshot ..." --> "The screenshot ..."
Add "a" before "... such run ..." --> "... such a run ..."

File: docs/dev/setup/local-database-tests.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+.. figure:: intellij-postgres-tests-run-configuration.png
+   :align: center
+   :alt: An example of IntelliJ run configuration fro starting Postgres server tests.
+
+
+Executing tests from console
```
Comment (JanaNF) [inline docs/dev/setup/local-database-tests.rst:31]: Add an article to "... console" --> "... the console"

File: docs/dev/setup/local-database-tests.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Database tests can be executed by running the dedicated Gradle command:
```
Comment (JanaNF) [inline docs/dev/setup/local-database-tests.rst:34]: Please consider not using passive voice and therefore rephrasing the "Database tests..." sentence.

**Review by JanaNF — CHANGES_REQUESTED** [review]
The documentation is easy to understand. 
I checked your documentation with Grammarly. Have a look at my comments and please consider changing some sentences.

File: docs/dev/setup/local-database-tests.rst [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+You can execute the database tests by running the run configurations provided in the repository:
+
+* For MySQL tests: `Artemis Server Tests MySQL`.
+* For Postgres tests: `Artemis Server Tests Postgres`.
+
+Or alternatively, by creating a custom configuration. The screenshot below presents an example of such a run configuration:
+
+.. figure:: intellij-postgres-tests-run-configuration.png
+   :align: center
+   :alt: An example of IntelliJ run configuration fro starting Postgres server tests.
```
Comment (sarpsahinalp) [inline docs/dev/setup/local-database-tests.rst:28]: Typo here in fro, should be from.

File: docs/dev/setup/local-database-tests.rst [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+You can execute the database tests by running the run configurations provided in the repository:
+
+* For MySQL tests: `Artemis Server Tests MySQL`.
+* For Postgres tests: `Artemis Server Tests Postgres`.
+
+Or alternatively, by creating a custom configuration. The screenshot below presents an example of such a run configuration:
+
+.. figure:: intellij-postgres-tests-run-configuration.png
+   :align: center
```
Comment (b-fein) [inline docs/dev/setup/local-database-tests.rst:27]:
```suggestion
   :align: center
   :scale: 50 %
```
The screenshot is pretty big and filly nearly the complete screen without scaling.

File: docs/dev/setup/local-database-tests.rst [hunk 8]
```diff
@@ -0,0 +1,37 @@
+.. _local_database_tests:
+
+Running database tests locally
+---------------------------------
+
+Prerequisites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Database tests are standard server integration tests executed against real database (MySQL or Postgres) started as Docker containers. Gradle test task will automatically pull necessary images and start Docker containers.
```
Comment (b-fein) [inline docs/dev/setup/local-database-tests.rst:9]:
```suggestion
Database tests are the standard server integration tests executed against a real database (MySQL or Postgres) started as Docker containers. The Gradle test task will automatically pull the necessary images and start the containers.
```

File: docs/dev/setup/local-database-tests.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+Before running those tests, make sure that the following steps were executed:
```
Comment (b-fein) [inline docs/dev/setup/local-database-tests.rst:11]:
```suggestion
Before running those tests, make sure you have got a working Docker setup:
```

File: docs/dev/setup/local-database-tests.rst [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+1. Install `Docker Desktop <https://docs.docker.com/desktop/#docker-for-mac>`__ or `Docker Engine and Docker CLI <https://docs.docker.com/engine/install/>`__.
```
Comment (b-fein) [inline docs/dev/setup/local-database-tests.rst:13]:
```suggestion
1. Install `Docker Desktop <https://docs.docker.com/desktop/>`__ or `Docker Engine and Docker CLI <https://docs.docker.com/engine/install/>`__.
```
Non-mac-specific link

File: docs/dev/setup/local-database-tests.rst [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+2. Enable `remote access for Docker daemon <https://docs.docker.com/config/daemon/remote-access/>`__.
```
Comment (b-fein) [inline docs/dev/setup/local-database-tests.rst:14]:
```suggestion
2. Enable `remote access for the Docker daemon <https://docs.docker.com/config/daemon/remote-access/>`__.
```

**Review by JanaNF — APPROVED** [review]

**Review by b-fein — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]
LGTM

**Review by sarpsahinalp — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**